presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsRecordCursor.java

```diff
@@ -0,0 +1,163 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.google.sheets;
+
+import com.google.common.base.Strings;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.prestosql.spi.connector.RecordCursor;
+import io.prestosql.spi.type.Type;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.DoubleType.DOUBLE;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+
+public class SheetsRecordCursor
+        implements RecordCursor
+{
+    private final List<SheetsColumnHandle> columnHandles;
+    private final int[] fieldToColumnIndex;
+    private final long totalBytes;
+    private final List<List<Object>> dataValues;
+
+    private List<String> fields;
+    private int currentIndex;
+
+    public SheetsRecordCursor(List<SheetsColumnHandle> columnHandles, List<List<Object>> dataValues)
+    {
+        this.columnHandles = columnHandles;
+        this.dataValues = dataValues;
+        fieldToColumnIndex = new int[columnHandles.size()];
+        for (int i = 0; i < columnHandles.size(); i++) {
+            SheetsColumnHandle columnHandle = columnHandles.get(i);
+            fieldToColumnIndex[i] = columnHandle.getOrdinalPosition();
+        }
+        long inputLength = 0;
+        for (List<Object> objList : dataValues) {
+            for (Object obj : objList) {
+                inputLength += String.valueOf(obj).length();
+            }
+        }
+        totalBytes = inputLength;
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return totalBytes;
+    }
+
+    @Override
+    public long getReadTimeNanos()
+    {
+        return 0;
+    }
+
+    @Override
+    public Type getType(int field)
+    {
+        checkArgument(field < columnHandles.size(), "Invalid field index");
+        return columnHandles.get(field).getColumnType();
+    }
+
+    @Override
+    public boolean advanceNextPosition()
+    {
+        List<Object> currentVals = null;
+        // Skip empty rows from sheet
+        while (currentVals == null || currentVals.size() == 0) {
+            if (currentIndex == dataValues.size()) {
+                return false;
+            }
+            currentVals = dataValues.get(currentIndex++);
+        }
+        // Populate incomplete columns with null
+        String[] allFields = new String[columnHandles.size()];
+
+        if (currentVals.size() > 0) {
```

Comment:
`currentVals.size() == 0` should never happen according to above while loop condition, you can replace this with checkstate